### PR TITLE
Devo v2 enh

### PR DESCRIPTION
**Pull request by manas-metron**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
- Added new parameter 'fields_to_view' to support the ability to control which fields in an alert or a query are sent to XSOAR.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by shmuel44 — COMMENTED** [review]
Hi @manas-metron,
Thank you for your contribution!
Good work :)

Please feel free to reach out to me with any questions
I'm available here or on slack :)
Thanks again

File: Packs/Devo/ReleaseNotes/1_3_0.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#### Integrations
+##### Devo v2
+- Added new parameter "fields_to_view" in devo-run-query, devo-get-alerts and devo-multi-table-query to filter out the specific fields in results.
```
Comment (shmuel44) [inline Packs/Devo/ReleaseNotes/1_3_0.md:3]:
```suggestion
- Added new argument ***fields_to_view*** to the commands ***devo-run-query***, ***devo-get-alerts*** and ***devo-multi-table-query*** to filter out the specific fields in results.
```

**Comment by shmuel44** [comment]
Hi @manas-metron, the code looks good!

We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.

**Comment by pavel-kalmykov-razgovorov** [comment]
Hi @shmuel44,

Apologies for the delay, but a customer has notified us they were having trouble setting up the `devo-write-to-lookup-table` command, and we noticed that both this and the `devo-write-to-table` commands were not working or not doing so correctly (respectively).

By analyzing what is needed to fix these issues, I realized the implementation of both commands needs to be drastically changed. Although we do not expect to modify any commands' parameters (their "API"), their behavior will differ. I am not sure whether you would prefer that we bump the upcoming version to a major one or get them included in this one.

In any case, what do you think would be best? To merge this PR and open a new one? Or can we have those changes included here altogether?

**Comment by shmuel44** [comment]
> In any case, what do you think would be best? To merge this PR and open a new one? Or can we have those changes included here altogether?

The new changes can be included in this PR.

**Comment by shmuel44** [comment]
Hi @manas-metron @pavel-kalmykov-razgovorov, 
Please change the PR status to "draft" until it's ready for code review.

**Comment by manas-metron** [comment]
Hello, @shmuel44  In light of the recent modifications, we would want to integrate this PR immediately. as it can take some time for the other modifications.
Our early time slot is 11 a.m., and we are in the IST time zone. We are prepared for the demo.

**Comment by manas-metron** [comment]
> > Hello, @shmuel44 In light of the recent modifications, we would want to integrate this PR immediately. as it can take some time for the other modifications. Our early time slot is 11 a.m., and we are in the IST time zone. We are prepared for the demo.
> 
> Hi @manas-metron, I have scheduled a meeting with @edik24 and you for tomorrow at 12:15 PM GMT timezone. Let me know if you didn't receive the invite.

Hi @shmuel44 I've received the invite. Thanks.

**Comment by shmuel44** [comment]
Hi @manas-metron, 
The code looks good!

As we discussed during the demo, I would appreciate it if you could clarify whether the `fields_to_view` argument should affect the `context data` or just the table (readable output) in the war room.
If it should affect the `context data`, please rename the argument to make its purpose more explicit. 

@edik24 FYI

**Comment by pavel-kalmykov-razgovorov** [comment]
Hi @shmuel44,

After talking about it with my team, we decided that it would be best to use a different name for this new parameter and keep the context and view aligned. We will rename it to `filter_in_columns`.

**Comment by shmuel44** [comment]
Hi @pavel-kalmykov-razgovorov, 
Perhaps use `filtered_columns`. 
Please add to the argument description that it also changes the `context data`.

**Comment by manas-metron** [comment]
> Hi @pavel-kalmykov-razgovorov, Perhaps use `filtered_columns`. Please add to the argument description that it also changes the `context data`.

Hello, @shmuel44 I modified the unit tests and changed the parameter name to `filtered_columns`. and change the parameter description as you suggested.

**Review by shmuel44 — APPROVED** [review]
Good Work!

**Comment by shmuel44** [comment]
Hi @manas-metron,
I merged the internal PR, and your update is now in the master branch of content repo.
Your update should be available in the marketplace within the next 12 hours.
Again, thank you very much for your contribution!

**Pull request merged**